Character.cpp

```diff
@@ -98,20 +97,33 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        //assert(false);
+        if(hitPoints < *initialHitPoints) hitPoints = *initialHitPoints;
+        if(armor < *initialArmorLevel) armor = *initialArmorLevel;
+        if(attackDamage < *initialAttackDamage) attackDamage = *initialAttackDamage;
+            //static_cast<int>( )
+        hitPoints *= 1.1;
+        armor *= 1.1;
+        attackDamage *= 1.1;
+
+        initialHitPoints.reset(new int(hitPoints));
+        initialArmorLevel.reset(new int(armor));
+        initialAttackDamage.reset(new int(attackDamage));
```

Comment:
1) D.R.Y.   you're doing the same thing to each pair of variables 3 times over. write a free function to update each pair of variables.
2) don't reset your initial hitpoints.  dereference and assign.
- [ ] revise